core/corehttp/routing.go

```diff
@@ -96,6 +96,57 @@ func (r *contentRouter) PutIPNS(ctx context.Context, name ipns.Name, record *ipn
 	return r.n.Routing.PutValue(ctx, string(name.RoutingKey()), raw)
 }
 
+func (r *contentRouter) GetClosestPeers(ctx context.Context, pid peer.ID) (iter.ResultIter[*types.PeerRecord], error) {
```

Comment:
If we decide to extend the spec to support lookup for the closest DHT servers to CIDs and IPNS keys (in addition to Peer IDs), we could accept a `string` instead of `peer.ID` as argument.

See https://github.com/ipfs/specs/pull/476#discussion_r2383524824

Reply:
updated to take cid.Cid